src/Raven.Server/Documents/Queries/Parser/QueryParser.cs

```diff
@@ -469,7 +473,9 @@ private bool GraphAlias(GraphQuery gq,bool isEdge, StringSegment implicitPrefix,
             if (Scanner.TryScan("WHERE"))
             {
                 if (Expression(out filter) == false)
-                    throw new InvalidQueryException($"Failed to parse filter expression for: {alias}", Scanner.Input, null);
+                    throw new InvalidQueryException($"There is a 'where' clause after '{collection.FieldValue}', but failed to parse the filter expression that should be after the 'where'. (The alias of the problematic query element is {alias})", Scanner.Input, null);
```

Comment:
Maybe:
`Failed to parse filter expression after a 'where' clause located next to '{collection.FieldValue}'. The problematic query element alias is '{alias}'`

Reply:
I like it